widget/entry.go

```diff
@@ -206,6 +303,86 @@ func (e *Entry) updateText(text string) {
 	Refresh(e)
 }
 
+// SelectionStart returns the first text position of the span of text covered by the selection highlight
+// eg: "T e s[t i]n g" == 3 (whitespace for clarity)
+func (e *Entry) SelectionStart() int {
```

Comment:
I don't think this method should be exported. SelectionStart and SelectionEnd accessed individually could be a race condition.

Reply:
Every API I've used provides a means to set and get the current selected region. I haven't yet provided setters.
Perhaps a better solution to avoid race conditions would be to provide a single function that returns start and end, eg:
```
func (e *Entry) GetSelection() int, int {
    ...
}
```

Reply:
Yes that would be better.
If we are unsure then this could be landed without the public APIs and we could add them later?
Better to get a smaller PR landed than to discuss at length without the main code in :)